### PR TITLE
Bug/fix of offer container

### PR DESCRIPTION
**Pull request by FryDay969**

### CODE REVIEW

File: src/containers/find-offer/offer-container/OfferContainer.tsx [hunk 1]
```diff
@@ -2,7 +2,8 @@ import { FC } from 'react'
 import Box from '@mui/material/Box'
 import { Grid } from '@mui/material'
```
Comment (Yurenko) [inline src/containers/find-offer/offer-container/OfferContainer.tsx:3]: Import like this:
import Grid from '@mui/material/Grid '

File: src/containers/find-offer/offer-container/OfferContainer.tsx [hunk 2]
```diff
@@ -2,7 +2,8 @@ import { FC } from 'react'
...
 import { CardsViewTypes, OfferResponse, CardsViewEnums } from '~/types'
-import OfferCardSquare from '../find-offer/offer-card-square/OfferCardSquare'
+import OfferCardSquare from '../offer-card-square/OfferCardSquare'
```
Comment (Yurenko) [inline src/containers/find-offer/offer-container/OfferContainer.tsx:5]: Import like this:
import OfferCard from '~/components/offer-card/OfferCard'
We use alias

File: src/containers/find-offer/offer-container/OfferContainer.tsx [hunk 3]
```diff
@@ -32,7 +24,23 @@ const OfferContainer: FC<OfferContainerProps> = ({ viewMode, offerCards }) => {
...
         spacing={{ md: 3 }}
       >
-        {arrayOfCards}
+        {viewMode === CardsViewEnums.Grid
+          ? offerCards.map((el) => (
+              <Grid item key={el._id} sm={4}>
+                <OfferCardSquare
+                  offer={el}
+                  onBookmarkClick={() => onBookmarkClick(el._id)}
+                />
```
Comment (Yurenko) [inline src/containers/find-offer/offer-container/OfferContainer.tsx:33]: Move out mapping from jsx

File: src/containers/find-offer/offer-container/OfferContainer.tsx [hunk 4]
```diff
@@ -32,7 +24,23 @@ const OfferContainer: FC<OfferContainerProps> = ({ viewMode, offerCards }) => {
...
+              </Grid>
+            ))
+          : offerCards.map((el) => (
+              <Grid item key={el._id} sm={4}>
+                <OfferCard
+                  offer={el}
+                  onBookmarkClick={() => onBookmarkClick(el._id)}
+                />
+              </Grid>
```
Comment (Yurenko) [inline src/containers/find-offer/offer-container/OfferContainer.tsx:42]: The same here

File: src/containers/find-offer/offer-container/OfferContainer.tsx (hunk 1 above)
Comment (FryDay969) [reply src/containers/find-offer/offer-container/OfferContainer.tsx:3]: Fixed.

File: src/containers/find-offer/offer-container/OfferContainer.tsx (hunk 2 above)
Comment (FryDay969) [reply src/containers/find-offer/offer-container/OfferContainer.tsx:5]: Fixed.

File: src/containers/find-offer/offer-container/OfferContainer.tsx (hunk 3 above)
Comment (FryDay969) [reply src/containers/find-offer/offer-container/OfferContainer.tsx:33]: Fixed.

File: src/containers/find-offer/offer-container/OfferContainer.tsx (hunk 4 above)
Comment (FryDay969) [reply src/containers/find-offer/offer-container/OfferContainer.tsx:42]: Fixed.

File: src/containers/find-offer/offer-container/OfferContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+const OfferContainer: FC<OfferContainerProps> = ({ viewMode, offerCards }) => {
+  const onBookmarkClick = (id: string) => {
+    console.log(id)
+  }
+
+  const columnNumber = viewMode === CardsViewEnums.Grid ? 12 : 1
+  const cardsArray =
+    viewMode === CardsViewEnums.Grid
+      ? offerCards.map((el) => (
```
Comment (Tolik170) [inline src/containers/find-offer/offer-container/OfferContainer.tsx:21]: dublicate of code, check for viewMode inside grid and render component you need or move this check to variable and pass this variable inside Grid

Comment (FryDay969) [reply src/containers/find-offer/offer-container/OfferContainer.tsx:21]: Discussed.

**Review by Tolik170 — APPROVED** [review]

**Review by MickhaelSh — COMMENTED** [review]
Add tests to your component

File: src/containers/find-offer/offer-container/OfferContainer.tsx (hunk 5 above)
Comment (MickhaelSh) [reply src/containers/find-offer/offer-container/OfferContainer.tsx:21]: i think Tolik is right, you should do:
offerCards.map((el) => (
          <Grid item key={el._id} sm={4}>
       viewMode === CardsViewEnums.Grid ?  <OfferCard
              offer={el}
              onBookmarkClick={() => onBookmarkClick(el._id)}
            /> :             <OfferCard
              offer={el}
              onBookmarkClick={() => onBookmarkClick(el._id)}
            />
          </Grid>

Comment (Tolik170) [reply src/containers/find-offer/offer-container/OfferContainer.tsx:21]: agree with Mykhailo, also, you can move this check to variable and pass it inside grid
const offerCard = viewMode === CardsViewEnums.Grid ? <OfferCard
offer={el}
onBookmarkClick={() => onBookmarkClick(el._id)}
/> : <OfferCard
offer={el}
onBookmarkClick={() => onBookmarkClick(el._id)}
/>

offerCards.map((el) => (
<Grid item key={el._id} sm={4}>
{offerCard}
</Grid>\

Comment (FryDay969) [reply src/containers/find-offer/offer-container/OfferContainer.tsx:21]: We have already reviewed this part of the code with Anatolii. Grid item requires filling in the "key" field. So, Grid item must be inside the array.

Comment (Tolik170) [reply src/containers/find-offer/offer-container/OfferContainer.tsx:21]: yeah, but here i leave Grid in map

Comment (FryDay969) [reply src/containers/find-offer/offer-container/OfferContainer.tsx]: Done.

File: src/containers/find-offer/offer-container/OfferContainer.tsx [hunk 6]
```diff
@@ -14,25 +15,29 @@ const OfferContainer: FC<OfferContainerProps> = ({ viewMode, offerCards }) => {
     console.log(id)
   }
 
-  const arrayOfCards = offerCards.map((el) => (
+  const columnNumber = viewMode === CardsViewEnums.Grid ? 12 : 1
```
Comment (MickhaelSh) [inline src/containers/find-offer/offer-container/OfferContainer.tsx:18]: Provide adaptivity for your component. it should show only inline on tablet and only square in mobile

File: src/containers/find-offer/offer-container/OfferContainer.tsx [hunk 7]
```diff
@@ -14,25 +15,29 @@ const OfferContainer: FC<OfferContainerProps> = ({ viewMode, offerCards }) => {
...
+
+  const cardsArray = offerCards.map((el) => (
```
Comment (MickhaelSh) [inline src/containers/find-offer/offer-container/OfferContainer.tsx:20]: it is not cardsArray, it is offerCards or offerItems

Comment (FryDay969) [reply src/containers/find-offer/offer-container/OfferContainer.tsx:20]: Fixed.

**Comment by FryDay969** [comment]
> Add tests to your component

File: src/containers/find-offer/offer-container/OfferContainer.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+const OfferContainer: FC<OfferContainerProps> = ({ viewMode, offerCards }) => {
+  const { isMobile, isDesktop } = useBreakpoints()
+  const onBookmarkClick = (id: string) => {
+    console.log(id)
+  }
+
+  const columnNumber = viewMode === CardsViewEnums.Grid ? 12 : 1
+
+  const renderSquareCard =
+    isMobile || (isDesktop && viewMode === CardsViewEnums.Grid) ? true : false
```
Comment (MickhaelSh) [inline src/containers/find-offer/offer-container/OfferContainer.tsx:23]: you don`t need ternary here, isMobile || (isDesktop && viewMode === CardsViewEnums.Grid), this expression returns boolean

File: src/containers/find-offer/offer-container/OfferContainer.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+      ) : (
+        <OfferCard offer={el} onBookmarkClick={() => onBookmarkClick(el._id)} />
+      )}
+    </Grid>
+  ))
+
+  return (
+    <Box data-testid='OfferContainer' sx={{ flexGrow: 1 }}>
+      <Grid
+        columns={{ sx: 1, sm: 1, md: columnNumber }}
```
Comment (MickhaelSh) [inline src/containers/find-offer/offer-container/OfferContainer.tsx:41]: there no sx. only xs, and when you use xs the same as sm you can omit using sm

File: src/containers/find-offer/offer-container/OfferContainer.tsx (hunk 8 above)
Comment (FryDay969) [reply src/containers/find-offer/offer-container/OfferContainer.tsx:23]: Cleaned

**Review by Yurenko — APPROVED** [review]

**Review by maarekua — APPROVED** [review]

**Pull request merged**